ProfitSite/src/main/java/bsu/fpmi/profit/CrudClass.java

```diff
@@ -0,0 +1,378 @@
+        ResultSet rsHashtags = reviewsStatement.executeQuery(queryHashtags);
+        while (rsHashtags.next()) {
+            hashTags.add(rsHashtags.getString("hashtag"));
+        }
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+        AdItem item = new AdItem(id, rs.getString("offer_name"),
+                rs.getString("description"),
+                sdf.parse(rs.getString("created_at")),
+                rs.getString("link"),
+                rs.getString("username"),
+                rs.getString("photo_link"),
+                hashTags,
+                rs.getString("discount"),
+                sdf.parse(rs.getString("valid_until")),
+                rating,
+                reviews);
+        rsReviews.close();
+        rsHashtags.close();
+        reviewsStatement.close();
+        hashtagsStatement.close();
+        return item;
+    }
+
+    public List<AdItem> getPage(AdFilters filters) {
+        int skip = computeSkip(filters);
+        int top = computeTop(filters);
+        String query = "SELECT *\n" +
+                "FROM profit.offer\n" +
+                "LEFT JOIN profit.user as user ON offer.USER_ID = user.USER_ID\n" +
+                prepareFilter(filters) +
+                "ORDER BY CREATED_AT DESC";
+        try {
+            PreparedStatement prStatement = prepareStatementFilters(filters, query);
+            ResultSet rs = prStatement.executeQuery();
+            List<AdItem> list = new ArrayList<AdItem>();
+            while (rs.next()) {
+                AdItem item = buildItem(rs);
+                list.add(item);
+            }
+            if (!filters.getHashTags().isEmpty()) {
+                list = list.stream().filter(ad -> ad.getHashTags().containsAll(filters.getHashTags()))
+                        .collect(Collectors.toList());
+            }
+            int count = skip + top;
+            rs.close();
+            prStatement.close();
+            if (count > list.size()) {
+                return list;
+            }
+            return list.subList(skip, count);
+        } catch (SQLException | ParseException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public AdItem get(String id) {
+        String query = "SELECT *\n" +
+                "FROM profit.offer\n" +
+                "LEFT JOIN profit.user as user ON offer.USER_ID = user.USER_ID\n" +
+                "WHERE OFFER_ID = ?";
+        try {
+            PreparedStatement prStatement = connection.prepareStatement(query);
+            prStatement.setInt(1, Integer.parseInt(id));
+            ResultSet rs = prStatement.executeQuery();
+            AdItem item = null;
+            if (rs.next()) {
+                item = buildItem(rs);
+            }
+            rs.close();
+            prStatement.close();
+            return item;
+        } catch (SQLException | ParseException e) {
+            e.printStackTrace();
```

Comment:
Outputting to stderr doesn't make much sense in a client-server application. Consider using a true logger, like log4j